src/en/rinkocomics/src/eu/kanade/tachiyomi/extension/en/rinkocomics/RinkoComics.kt

```diff
@@ -0,0 +1,400 @@
+        parseComicsPage(response)
+
+    override fun mangaDetailsParse(response: Response): SManga =
+        parseMangaDetails(response.asJsoup())
+
+    private fun parseMangaDetails(document: Document): SManga = SManga.create().apply {
+        title = requireField(
+            document.selectFirst(".comic-info-upper h1")?.text()
+                ?: document.selectFirst("h1")?.text(),
+            "title",
+        )
+
+        thumbnail_url = document.selectFirst("meta[property=og:image]")?.attr("content")
+
+        val authors = document.select(".comic-graph > span")
+            .map { it.text().trim() }
+            .filter { it.isNotBlank() && it != "•" }
+            .distinct()
+
+        author = authors.firstOrNull()
+        artist = authors.getOrNull(1)
+
+        status = parseStatus(
+            document.selectFirst(".comic-status span:last-child")?.text(),
+        )
+
+        genre = document.select(".comic-genres .genres .genre")
+            .joinToString(", ") { it.text().trim() }
+
+        description = document.selectFirst(".comic-synopsis")?.text()?.trim()
+
+        initialized = true
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val document = response.asJsoup()
+        val hideLocked = preferences.getBoolean(PREF_HIDE_LOCKED, false)
+
+        val chapters = LinkedHashMap<String, SChapter>()
+        fun addAll(items: List<SChapter>) {
+            items.forEach { chapter ->
+                if (!chapters.containsKey(chapter.url)) {
+                    chapters[chapter.url] = chapter
+                }
+            }
+        }
+
+        addAll(parseChapterElements(document.select(CHAPTER_SELECTOR), hideLocked))
+
+        val loadMoreBtn = document.selectFirst("#loadMoreChaptersBtn")
+        val comicId = loadMoreBtn?.attr("data-comic-id").orEmpty()
+        val nonce = extractNonce(document).orEmpty()
+        var offset = loadMoreBtn?.attr("data-offset")?.toIntOrNull() ?: 0
+        if (offset <= 0) {
+            offset = chapters.size
+        } else if (chapters.isNotEmpty() && offset > chapters.size) {
+            offset = chapters.size
+        }
+
+        if (comicId.isNotBlank() && nonce.isNotBlank()) {
+            while (true) {
+                val items = fetchMoreChapters(comicId, offset, nonce, hideLocked)
+                if (items.isEmpty()) break
+                addAll(items)
+                offset += CHAPTERS_PER_PAGE
+            }
+        }
+
+        return chapters.values.toList()
+    }
+
+    override fun pageListParse(response: Response): List<Page> {
+        val document = response.asJsoup()
+        val pages = document.select("img.chapter-image").mapIndexedNotNull { index, element ->
+            val imageUrl = element.attr("abs:data-src").ifBlank { element.attr("abs:src") }.trim()
+            if (imageUrl.isBlank()) return@mapIndexedNotNull null
+            Page(index, document.location(), imageUrl)
```

Comment:
If you don't use the page URL later you can omit it.
```suggestion
            Page(index, imageUrl = imageUrl)
```